gateway-js/src/loadServicesFromRemoteEndpoint.ts

```diff
@@ -3,20 +3,22 @@ import { parse } from 'graphql';
 import { Headers, HeadersInit } from 'node-fetch';
 import { GraphQLDataSource } from './datasources/types';
 import { SERVICE_DEFINITION_QUERY } from './';
-import { CompositionUpdate } from './config';
+import { CompositionUpdate, ServiceEndpointDefinition } from './config';
 import { ServiceDefinition } from '@apollo/federation';
 
+type Service = ServiceEndpointDefinition & {
+  dataSource: GraphQLDataSource;
+};
+
 export async function getServiceDefinitionsFromRemoteEndpoint({
   serviceList,
-  headers = {},
+  getServiceIntrospectionHeaders,
   serviceSdlCache,
 }: {
-  serviceList: {
-    name: string;
-    url?: string;
-    dataSource: GraphQLDataSource;
-  }[];
-  headers?: HeadersInit;
+  serviceList: Service[];
+  getServiceIntrospectionHeaders?: (
```

Comment:
I'm unsure why `headers` was previously optional, but I only see one callsite for this function and `getServiceIntrospectionHeaders` is provided so I think we can remove the `?` in the signature and subsequently where we actually call it.

Reply:
I had the same thought tbh, but didn't want to change things unnecessarily. Will update :+1: